src/Integration.UnitTests/Integration.UnitTests.csproj

```diff
@@ -7,10 +7,10 @@
     <RequiresRuleset>true</RequiresRuleset>
     <RequiresAlmAuthentication>true</RequiresAlmAuthentication>
     <RequiresNewtonsoft>true</RequiresNewtonsoft>
-    <!-- Conditional compilation constant for VS2019-specific code.
+    <!-- Conditional compilation constant for VS2022-specific code.
          We'd prefer to set this the shared SonarLint.VSSpecificAssemblies.props file, but
          the IDE isn't picking up the setting if we do that. -->
-    <DefineConstants Condition="($(VisualStudioVersion)=='17.0' AND $(BuildingInsideVisualStudio)=='true') OR $(VSTargetVersion)=='2022'">$(DefineConstants);VS2022</DefineConstants>
+    <DefineConstants Condition="$(VSTargetVersion)=='2022'">$(DefineConstants);VS2022</DefineConstants>
```

Comment:
Miscellaneous: the condition can be simplified since the '$(VSTargetVersion)' is now guaranteed to be set inside and outside VS (it's done via the `Import SonarLint.Test.props` in line 3).

I've checked this works in the VS2017 and VS2022 IDEs.